vllm/entrypoints/nixl_side_channel_server.py

```diff
@@ -0,0 +1,123 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright contributors to the vLLM project
+
+import asyncio
+from typing import Any, Optional
+
+import uvicorn
+from fastapi import FastAPI, HTTPException
+
+from vllm import envs
+from vllm.config import VllmConfig
+from vllm.logger import init_logger
+
+logger = init_logger(__name__)
+
+
+class NixlSideChannelServer:
```

Comment:
As mentioned elsewhere we should make this "kv connector generic" if it lives in the upper layers